/*
 * Copyright (C) 2023 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package cuttlefish.run_cvd;

message ExtendedLauncherAction {
  oneof actions {
    Suspend suspend = 5;
    Resume resume = 6;
    StartScreenRecording start_screen_recording = 7;
    StopScreenRecording stop_screen_recording = 8;
    SnapshotTake snapshot_take = 9;
  }
  string verbosity = 20;
}
message Suspend {}
message Resume {}
message StartScreenRecording {}
message StopScreenRecording {}
message SnapshotTake {
  repeated string snapshot_path = 1;
}
